packages/@aws-cdk/aws-apigateway/lib/integration.ts

```diff
@@ -0,0 +1,260 @@
+import iam = require('@aws-cdk/aws-iam');
+import { Method } from './method';
+
+export interface IntegrationOptions {
+    /**
+     * A list of request parameters whose values API Gateway caches.
+     */
+    cacheKeyParameters?: string[];
+
+    /**
+     * An API-specific tag group of related cached parameters.
+     */
+    cacheNamespace?: string;
+
+    /**
+     * Specifies how to handle request payload content type conversions.
+     *
+     * @default none if this property isn't defined, the request payload is passed
+     * through from the method request to the integration request without
+     * modification, provided that the `passthroughBehaviors` property is
+     * configured to support payload pass-through.
+     */
+    contentHandling?: ContentHandling;
+
+    /**
+     * An IAM role that API Gateway assumes.
+     *
+     * Mutually exclusive with `credentialsPassThrough`.
+     *
+     * @default A role is not assumed
+     */
+    credentialsRole?: iam.Role;
+
+    /**
+     * Requires that the caller's identity be passed through from the request.
+     *
+     * @default Caller identity is not passed through
+     */
+    credentialsPassthrough?: boolean;
+
+    /**
+     * Specifies the pass-through behavior for incoming requests based on the
+     * Content-Type header in the request, and the available mapping templates
+     * specified as the requestTemplates property on the Integration resource.
+     * There are three valid values: WHEN_NO_MATCH, WHEN_NO_TEMPLATES, and
+     * NEVER.
+     */
+    passthroughBehavior?: PassthroughBehavior
+
+    /**
+     * The request parameters that API Gateway sends with the backend request.
+     * Specify request parameters as key-value pairs (string-to-string
+     * mappings), with a destination as the key and a source as the value.
+     *
+     * Specify the destination by using the following pattern
+     * integration.request.location.name, where location is querystring, path,
+     * or header, and name is a valid, unique parameter name.
+     *
+     * The source must be an existing method request parameter or a static
+     * value. You must enclose static values in single quotation marks and
```

Comment:
This as well is begging for an improved developer experience...

Reply:
I am not even sure how much people use these features...